### PR TITLE
Initial version of ElasticSearch mixin

### PR DESCRIPTION
**Pull request by colega**
Ref: https://github.com/grafana/integrations-pipeline/issues/33

Based on this dashboard: https://grafana.com/grafana/dashboards/2322

You can check out a snapshot of it: https://snapshot.raintank.io/dashboard/snapshot/vBLYWY907sezsJBBdcKlNlAaWtu7ZKV1

It looks like this:

image

### CODE REVIEW

**Review by sdboyer — CHANGES_REQUESTED** [review]
Let's have both `$job` and `$instance` in the queries, as both of those are important for differentiating data in various possible setups.

Also, just using those variable names directly in the template variables (so "instance" instead of "Source of metrics") is preferable, for consistency with other mixins.

**Comment by colega** [comment]
@sdboyer added $job to all queries, removed label names overrides, so instance is now instance in https://github.com/grafana/jsonnet-libs/pull/427/commits/0aad3b0c15067b27416634d8a810a32dcdf54fed 

PTAL.

File: elasticsearch-mixin/dashboards/elasticsearch-overview.json [hunk 1]
```diff
@@ -0,0 +1,6478 @@
...
+          "pointradius": 5,
+          "points": false,
+          "renderer": "flot",
+          "seriesOverrides": [],
+          "spaceLength": 10,
+          "stack": false,
+          "steppedLine": false,
+          "targets": [
+            {
+              "expr": "irate(elasticsearch_thread_pool_completed_count{job=\"$job\",cluster=\"$cluster\",name=~\"$name\"}[$interval])",
```
Comment (sdboyer) [inline elasticsearch-mixin/dashboards/elasticsearch-overview.json:4677]: We should probably prefer the [new](https://grafana.com/blog/2020/09/28/new-in-grafana-7.2-__rate_interval-for-prometheus-rate-queries-that-just-work/) `$__rate_interval` here (and elsewhere throughout)

File: elasticsearch-mixin/dashboards/elasticsearch-overview.json [hunk 2]
```diff
@@ -0,0 +1,6478 @@
...
+          "pointradius": 5,
+          "points": false,
+          "renderer": "flot",
+          "seriesOverrides": [],
+          "spaceLength": 10,
+          "stack": false,
+          "steppedLine": false,
+          "targets": [
+            {
+              "expr": "irate(elasticsearch_jvm_gc_collection_seconds_count{job=\"$job\",cluster=\"$cluster\",name=~\"$name\"}[$interval])",
```
Comment (sdboyer) [inline elasticsearch-mixin/dashboards/elasticsearch-overview.json:1502]: A number of the queries don't seem to have the `$instance` variable - is that intentional?

**Review by sdboyer — CHANGES_REQUESTED** [review]
Couple more things, sorry to do it piecemeal!

File: elasticsearch-mixin/dashboards/elasticsearch-overview.json (hunk 2 above)
Comment (colega) [reply elasticsearch-mixin/dashboards/elasticsearch-overview.json]: Nice catch, I guess the original author of the dashboard relied just on the $cluster label. Added $instance everywhere in https://github.com/grafana/jsonnet-libs/pull/427/commits/196ef248766a1aa5bcfcfb9ea6d7691043235afd

File: elasticsearch-mixin/dashboards/elasticsearch-overview.json (hunk 1 above)
Comment (colega) [reply elasticsearch-mixin/dashboards/elasticsearch-overview.json]: TIL and used in https://github.com/grafana/jsonnet-libs/pull/427/commits/888f0df2dc5d4060604376bd0f2bf684882cafb0

**Comment by colega** [comment]
Ready to be reviewed again :)

**Comment by colega** [comment]
For your convenience, here's a snapshot of the latest version of the dashboard: https://snapshot.raintank.io/dashboard/snapshot/AMGvl8PkW1Jd9r6fgioSaRgeNKfP7501

**Review by sdboyer — APPROVED** [review]
LGTM!

**Pull request merged**